deploy-board/deploy_board/webapp/cluster_view.py

```diff
@@ -646,7 +646,9 @@ def clone_cluster(request, src_name, src_stage):
         ##0. teletraan service get src env buildName
         src_env = environs_helper.get_env_by_stage(request, src_name, src_stage)
         build_name = src_env.get('buildName', None)
-        external_id = src_env.get('externalId', None)
+        external_id = environs_helper.create_identifier_for_new_stage(request, dest_name, dest_stage)
+        if external_id is None:
+            raise TeletraanException("Unable to find Nimbus identifier for enviornment %s" % dest_name)
```

Comment:
this is a different behavior from existing flow which is not failing on nimbus id creation.

Also, if stage creation in Teletraan fails, do we clean up generated nimbus id?

Reply:
^ can be wrapped in helper method to avoid repeated code.

Reply:
Thanks, updated
* Added `IS_PINTEREST` and `identifier.get('uuid')` check to helper method
* Added cleanup for nimbus id